### PR TITLE
Update XLM with TF decorator

### PR DESCRIPTION
**Pull request by louisowen6**
# What does this PR do?

Unpacks TF model inputs through a decorator, improving code clarity for the `XLM` model. To be read with issue https://github.com/huggingface/transformers/issues/16051 , which holds the description of the problem, the proposed solution, and future plan.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)?  https://github.com/huggingface/transformers/issues/16051
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@LysandreJik @gante

### CODE REVIEW

**Comment by louisowen6** [comment]
tests results

```
RUN_SLOW=1 py.test -vv tests/xlm/test_modeling_tf_xlm.py      
================================================================================== test session starts ===================================================================================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0 -- /home/louisowen6/miniconda3/envs/transformers_dev/bin/python3
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/mnt/c/Users/Louis Owen/transformers/.hypothesis/examples')
rootdir: /mnt/c/Users/Louis Owen/transformers, configfile: setup.cfg
plugins: hypothesis-6.39.4, forked-1.4.0, timeout-2.1.0, xdist-2.5.0, dash-2.3.0
collected 36 items                                                                                                                                                                       

tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_attention_outputs <- tests/test_modeling_tf_common.py PASSED                                                               [  2%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_compile_tf_model <- tests/test_modeling_tf_common.py PASSED                                                                [  5%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_config PASSED                                                                                                              [  8%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_determinism <- tests/test_modeling_tf_common.py PASSED                                                                     [ 11%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_for_multiple_choice PASSED                                                                                                 [ 13%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_for_token_classification PASSED                                                                                            [ 16%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_forward_signature <- tests/test_modeling_tf_common.py PASSED                                                               [ 19%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_generate_with_headmasking <- tests/test_modeling_tf_common.py PASSED                                                       [ 22%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_headmasking <- tests/test_modeling_tf_common.py PASSED                                                                     [ 25%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_hidden_states_output <- tests/test_modeling_tf_common.py PASSED                                                            [ 27%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_initialization <- tests/test_modeling_tf_common.py PASSED                                                                  [ 30%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_inputs_embeds <- tests/test_modeling_tf_common.py PASSED                                                                   [ 33%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_keras_save_load <- tests/test_modeling_tf_common.py PASSED                                                                 [ 36%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_keyword_and_dict_args <- tests/test_modeling_tf_common.py PASSED                                                           [ 38%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_beam_search_generate_dict_outputs <- tests/test_modeling_tf_common.py PASSED                                 [ 41%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_no_beam_search_generate_dict_outputs <- tests/test_modeling_tf_common.py PASSED                              [ 44%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_random_beam_search_generate <- tests/test_modeling_tf_common.py PASSED                                       [ 47%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_random_no_beam_search_generate <- tests/test_modeling_tf_common.py PASSED                                    [ 50%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_load_with_mismatched_shapes <- tests/test_modeling_tf_common.py PASSED                                                     [ 52%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_loss_computation <- tests/test_modeling_tf_common.py PASSED                                                                [ 55%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_common_attributes <- tests/test_modeling_tf_common.py PASSED                                                         [ 58%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_from_pretrained PASSED                                                                                               [ 61%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_main_input_name <- tests/test_modeling_tf_common.py PASSED                                                           [ 63%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_outputs_equivalence <- tests/test_modeling_tf_common.py PASSED                                                       [ 66%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_numpy_arrays_inputs <- tests/test_modeling_tf_common.py PASSED                                                             [ 69%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_onnx_compliancy <- tests/test_modeling_tf_common.py PASSED                                                                 [ 72%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_onnx_runtime_optimize <- tests/test_modeling_tf_common.py PASSED                                                           [ 75%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_pt_tf_model_equivalence <- tests/test_modeling_tf_common.py SKIPPED (test is PT+TF test)                                   [ 77%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_resize_token_embeddings <- tests/test_modeling_tf_common.py PASSED                                                         [ 80%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_save_load <- tests/test_modeling_tf_common.py PASSED                                                                       [ 83%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_save_load_config <- tests/test_modeling_tf_common.py PASSED                                                                [ 86%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_lm_head PASSED                                                                                                         [ 88%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_model PASSED                                                                                                           [ 91%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_qa PASSED                                                                                                              [ 94%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_sequence_classif PASSED                                                                                                [ 97%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelLanguageGenerationTest::test_lm_generate_xlm_mlm_en_2048 PASSED                                                                       [100%]
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 1]
```diff
@@ -708,6 +691,7 @@ def __init__(self, config, *inputs, **kwargs):
         output_type=TFBaseModelOutput,
         config_class=_CONFIG_FOR_DOC,
     )
+    @unpack_inputs
```
Comment (gante) [inline src/transformers/models/xlm/modeling_tf_xlm.py:694]: Apologies for being picky about this one -- other PRs have been adding the decorator as the top decorator. Can you please move it, so it stays consistent with other models? 🙏  (there are more instances of this in the file)

**Review by gante — APPROVED** [review]
Awesome, thank you so much for the contribution 🔥  Have a look at my comment, should be quick to attend to

File: src/transformers/models/xlm/modeling_tf_xlm.py (hunk 1 above)
Comment (louisowen6) [reply src/transformers/models/xlm/modeling_tf_xlm.py]: Sure thing. I've moved it as top decorator :)

Comment (gante) [reply src/transformers/models/xlm/modeling_tf_xlm.py]: There are at least two more instances of `@unpack_inputs` not being the top decorator -- can you change them as well? :)

Comment (gante) [reply src/transformers/models/xlm/modeling_tf_xlm.py]: (and don't forget the `make fixup` before committing)

Comment (louisowen6) [reply src/transformers/models/xlm/modeling_tf_xlm.py]: Ahh sure, apologize for this. Will add the rest and run `make fixup` before committing 👍

**Comment by louisowen6** [comment]
second tests report

```
 RUN_SLOW=1 py.test -vv tests/xlm/test_modeling_tf_xlm.py
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0 -- /home/louisowen6/miniconda3/envs/transformers_dev/bin/python3
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/mnt/c/Users/Louis Owen/Desktop/transformers/.hypothesis/examples')
rootdir: /mnt/c/Users/Louis Owen/Desktop/transformers, configfile: setup.cfg
plugins: hypothesis-6.39.4, forked-1.4.0, timeout-2.1.0, xdist-2.5.0, dash-2.3.0
collected 36 items                                                                                                                                                                 

tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_attention_outputs <- tests/test_modeling_tf_common.py PASSED                                                         [  2%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_compile_tf_model <- tests/test_modeling_tf_common.py PASSED                                                          [  5%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_config PASSED                                                                                                        [  8%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_determinism <- tests/test_modeling_tf_common.py PASSED                                                               [ 11%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_for_multiple_choice PASSED                                                                                           [ 13%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_for_token_classification PASSED                                                                                      [ 16%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_forward_signature <- tests/test_modeling_tf_common.py PASSED                                                         [ 19%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_generate_with_headmasking <- tests/test_modeling_tf_common.py PASSED                                                 [ 22%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_headmasking <- tests/test_modeling_tf_common.py PASSED                                                               [ 25%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_hidden_states_output <- tests/test_modeling_tf_common.py PASSED                                                      [ 27%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_initialization <- tests/test_modeling_tf_common.py PASSED                                                            [ 30%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_inputs_embeds <- tests/test_modeling_tf_common.py PASSED                                                             [ 33%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_keras_save_load <- tests/test_modeling_tf_common.py PASSED                                                           [ 36%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_keyword_and_dict_args <- tests/test_modeling_tf_common.py PASSED                                                     [ 38%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_beam_search_generate_dict_outputs <- tests/test_modeling_tf_common.py PASSED                           [ 41%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_no_beam_search_generate_dict_outputs <- tests/test_modeling_tf_common.py PASSED                        [ 44%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_random_beam_search_generate <- tests/test_modeling_tf_common.py PASSED                                 [ 47%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_lm_head_model_random_no_beam_search_generate <- tests/test_modeling_tf_common.py PASSED                              [ 50%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_load_with_mismatched_shapes <- tests/test_modeling_tf_common.py PASSED                                               [ 52%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_loss_computation <- tests/test_modeling_tf_common.py PASSED                                                          [ 55%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_common_attributes <- tests/test_modeling_tf_common.py PASSED                                                   [ 58%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_from_pretrained PASSED                                                                                         [ 61%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_main_input_name <- tests/test_modeling_tf_common.py PASSED                                                     [ 63%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_model_outputs_equivalence <- tests/test_modeling_tf_common.py PASSED                                                 [ 66%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_numpy_arrays_inputs <- tests/test_modeling_tf_common.py PASSED                                                       [ 69%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_onnx_compliancy <- tests/test_modeling_tf_common.py PASSED                                                           [ 72%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_onnx_runtime_optimize <- tests/test_modeling_tf_common.py PASSED                                                     [ 75%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_pt_tf_model_equivalence <- tests/test_modeling_tf_common.py SKIPPED (test is PT+TF test)                             [ 77%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_resize_token_embeddings <- tests/test_modeling_tf_common.py PASSED                                                   [ 80%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_save_load <- tests/test_modeling_tf_common.py PASSED                                                                 [ 83%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_save_load_config <- tests/test_modeling_tf_common.py PASSED                                                          [ 86%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_lm_head PASSED                                                                                                   [ 88%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_model PASSED                                                                                                     [ 91%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_qa PASSED                                                                                                        [ 94%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelTest::test_xlm_sequence_classif PASSED                                                                                          [ 97%]
tests/xlm/test_modeling_tf_xlm.py::TFXLMModelLanguageGenerationTest::test_lm_generate_xlm_mlm_en_2048 PASSED                                                                 [100%]
```

**Comment by gante** [comment]
Perfect, thanks! Will merge as soon as CI gets to green

**Pull request merged**